kernel/src/common/utils.rs

```diff
@@ -81,3 +81,23 @@ macro_rules! create_capability {
         Cap
     };};
 }
+
+
+/// Creates a static object with the given capability.
+/// Works as create_capability above, but the given capability has a static
+/// lifetime.
+#[macro_export]
+macro_rules! create_static_capability {
```

Comment:
Do we need both of these? Are there times when only one will work? I think this will be confusing for developers (or at least for me) who then have to decide which to use.

Reply:
The static capability is essential when we need a capability to have a static lifetime -- these are for capabilities that capsules will need to hold onto as opposed to capabilities that are just passed once to functions.

Reply:
Gotcha. Is there a way to just have one macro? Can we just always use the static version?

Reply:
That's a good question -- wouldn't there be an additional cost with having to tag a piece of memory as static that wouldn't otherwise need to be?

Reply:
Furthermore, wouldn't we want the flexibility to specify what capabilities should have what lifetimes?

Reply:
I personally would find this flexibility confusing, and want something that "just works". But, I don't understand the potential tradeoffs and effects (correctness, code size, etc) of the available options.